cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataHttpHandlerTest.java

```diff
@@ -85,6 +87,26 @@ public void testProperties() throws IOException {
     Assert.assertEquals(datasetProperties, properties);
     properties = getProperties(mystream);
     Assert.assertEquals(streamProperties, properties);
+
+    // test search for stream
+    Set<MetadataSearchResultRecord> searchProperties = searchMetadata(Id.Namespace.DEFAULT.getId(),
+                                                                      "stKey:stValue", "STREAM");
+    Set<MetadataSearchResultRecord> expected = ImmutableSet.of(
+      new MetadataSearchResultRecord(mystream, MetadataSearchTargetType.STREAM)
+    );
+    Assert.assertEquals(expected, searchProperties);
+
+    // test prefix search for service
+    searchProperties = searchMetadata(Id.Namespace.DEFAULT.getId(), "sKey:s", "ALL");
+    expected = ImmutableSet.of(
+      new MetadataSearchResultRecord(pingService, MetadataSearchTargetType.PROGRAM)
+    );
+    Assert.assertEquals(expected, searchProperties);
+
+    // search without any target param
+    searchProperties = searchMetadata(Id.Namespace.DEFAULT.getId(), "sKey:s", null);
+    Assert.assertEquals(expected, searchProperties);
+
```

Comment:
in both these search verifications, can you add a test that returns empty? a good case would be to search for an existing property/tag, but on the wrong target type.